File: src/Complex/Prog2/Forcedresponse/Forcedresponse.java
package Complex.Prog2.Forcedresponse;

import java.applet.Applet;
import java.awt.BorderLayout;
import java.awt.Image;
import java.awt.Label;
import java.awt.Panel;

public class Forcedresponse extends Applet implements Runnable {
   static final double kMinL = 0.5;
   static final double kMaxL = 2.0;
   static final double kMinD = 0.05;
   static final double kMaxD = 0.2;
   static final double kMinm = 5.0;
   static final double kMaxm = 100.0;
   static final String kRunString = "Run";
   CFramePanel mFramePanel;
   CFrameAnimation mAnimFrame;
   Thread mThread = null;

   public void init() {
      this.setLayout(new BorderLayout());
      Panel buttonPanel = new Panel();
      this.add("North", buttonPanel);
      Label copyright = new Label("  This program is copyright, you need permission to use it. © 2004 B J Stone, bjs@mech.uwa.edu.au.");
      this.add("South", copyright);
      this.mFramePanel = new CFramePanel(this);
      this.add("Center", this.mFramePanel);
      this.mAnimFrame = new CFrameAnimation(this.mFramePanel, 0, 0, this.size().width, this.size().height);
      CFrameSmallControl LControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 1, 480, 310, 0.5, 1.0, 2.0, "m");
      LControl.SetLabelOffset(20, 10);
      LControl.SetTextEditable(true);
      Image LImage = this.getImage(this.getCodeBase(), "L.gif");
      new CFramePicture(this.mFramePanel, 500, 325, 0, 0, LImage, false);
      CFrameSmallControl DControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 2, 570, 310, 0.05, 0.1, 0.2, "m");
      DControl.SetLabelOffset(20, 10);
      DControl.SetTextEditable(true);
      Image DImage = this.getImage(this.getCodeBase(), "D.gif");
      new CFramePicture(this.mFramePanel, 590, 325, 0, 0, DImage, false);
      CFrameSmallControl mControl = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 3, 660, 310, 5.0, 20.0, 100.0, "kg");
      mControl.SetLabelOffset(20, 10);
      mControl.SetTextEditable(true);
      Image mImage = this.getImage(this.getCodeBase(), "m.gif");
      new CFramePicture(this.mFramePanel, 680, 325, 0, 0, mImage, false);
      Image DiagramImage = this.getImage(this.getCodeBase(), "Diagram.gif");
      new CFramePicture(this.mFramePanel, 470, 30, 0, 0, DiagramImage, false);
      Image XaxisImage = this.getImage(this.getCodeBase(), "Xaxis.gif");
      new CFramePicture(this.mFramePanel, 85, 315, 0, 0, XaxisImage, false);
      Image YaxisImage = this.getImage(this.getCodeBase(), "Yaxis.gif");
      new CFramePicture(this.mFramePanel, 1, 3, 0, 0, YaxisImage, false);
      this.validate();
   }

   public void start() {
      this.mThread = new Thread(this);
      this.mThread.start();
   }

   public void run() {
      while (true) {
         try {
            Thread.sleep(20L);
         } catch (InterruptedException var1) {
            return;
         }
      }
   }
}

File: site/swingjs/j2s/OneDOF/Prog6/Forcedresponse/Forcedresponse.js
(function(){var P$=Clazz.newPackage("OneDOF.Prog6.Forcedresponse"),I$=[[0,'java.awt.BorderLayout','java.awt.Panel','java.awt.Label','OneDOF.Prog6.Forcedresponse.CFramePanel','OneDOF.Prog6.Forcedresponse.CFrameAnimation','OneDOF.Prog6.Forcedresponse.CFrameSmallControl','OneDOF.Prog6.Forcedresponse.CFramePicture','javax.swing.Timer']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Forcedresponse", null, 'java.applet.Applet', 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.mTimer=null;
this.mLastTime=0;
},1);

C$.$fields$=[['J',['mLastTime'],'O',['mFramePanel','OneDOF.Prog6.Forcedresponse.CFramePanel','mAnimFrame','OneDOF.Prog6.Forcedresponse.CFrameAnimation','mTimer','javax.swing.Timer']]]

Clazz.newMeth(C$, ['init$','init'],  function () {
this.setLayout$java_awt_LayoutManager(Clazz.new_($I$(1,1)));
var buttonPanel=Clazz.new_($I$(2,1));
this.add$S$java_awt_Component("North", buttonPanel);
var copyright=Clazz.new_($I$(3,1).c$$S,["  This program is copyright, you need permission to use it. \u00a9 1999 B J Stone, bjs@mech.uwa.edu.au."]);
this.add$S$java_awt_Component("South", copyright);
this.mFramePanel=Clazz.new_($I$(4,1).c$$java_applet_Applet,[this]);
this.add$S$java_awt_Component("Center", this.mFramePanel);
this.mAnimFrame=Clazz.new_([this.mFramePanel, 0, 0, this.getSize$().width, this.getSize$().height],$I$(5,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I);
var XiControl=Clazz.new_($I$(6,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$OneDOF_Prog6_Forcedresponse_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 1, 100, 10, 0.001, 0.1, 3.0, " "]);
XiControl.SetLabelOffset$I$I(20, 10);
XiControl.SetTextEditable$Z(true);
var XiImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "xi.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 128, 25, 0, 0, XiImage, false]);
var DiagramImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Diagram.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 460, 60, 0, 0, DiagramImage, false]);
var XaxisImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Xaxis.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 97, 220, 0, 0, XaxisImage, false]);
var YaxesImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Yaxes.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 62, 45, 0, 0, YaxesImage, false]);
var kXdivFImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "kXdivF.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 105, 65, 0, 0, kXdivFImage, false]);
var phiImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "phi.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 105, 320, 0, 0, phiImage, false]);
var WdivWnImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "WdivWn.gif");
Clazz.new_($I$(7,1).c$$OneDOF_Prog6_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 412, 215, 0, 0, WdivWnImage, false]);
this.validate$();
});

Clazz.newMeth(C$, ['start$','start'],  function () {
this.mTimer=Clazz.new_($I$(8,1).c$$I$java_awt_event_ActionListener,[20, this]);
this.mTimer.start$();
});

Clazz.newMeth(C$, ['actionPerformed$java_awt_event_ActionEvent','actionPerformed'],  function (e) {
if (this.mTimer != null ) {
var ddd=Clazz.new_(java.util.Date);
var thisTime=ddd.getTime$();
if (Long.$eq(this.mLastTime,0 )) {
this.mLastTime=thisTime;
}this.mLastTime=thisTime;
}});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:51 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
